File: src/Maximum 69 Number/main.cpp
#include <vector>

class Solution {
public:
    int maximum69Number (int num) {
        std::vector<int> stack;
        while (num > 0) {
            stack.push_back(num % 10);
            num /= 10;
        }

        for (int i = stack.size() - 1; i >= 0; --i) {
            if (stack[i] == 9) continue;
            stack[i] = 9;
            break;
        }

        num = 0;
        for (int i = stack.size() - 1; i >= 0; --i) num = num * 10 + stack[i];
        return num;
    }
};

File: src/Maximum 69 Number/main.py
from typing import List

class Solution:
    def maximum69Number (self, num: int) -> int:
        stack: List[int] = []
        while num > 0:
            stack.append(num % 10)
            num //= 10
        stack = stack[::-1]
        for i in range(len(stack)):
            if stack[i] == 6:
                stack[i] = 9
                break
        answer = 0
        for i in range(len(stack)): answer = answer * 10 + stack[i]
        return answer